scripts/jenkins-yoctobuild-build.sh

```diff
@@ -2,19 +2,24 @@
 
 set -e -x -E
 
-$WORKSPACE/mender-qa/scripts/wait-until-build-host-ready.sh
+if [ -n "$JENKINS_URL" ]; then
+    $WORKSPACE/mender-qa/scripts/wait-until-build-host-ready.sh
+fi
 
 echo $WORKSPACE
 
-sudo apt-get -qy --force-yes install docker-ce || apt-get -qy --force-yes install docker-ce
-# make sure Docker uses the block storage to store docker containers.
-sudo bash -c 'cat << EOF > /etc/docker/daemon.json
-{
-"data-root": "/var/lib/docker/",
-"storage-driver": "overlay2"
-}
-EOF'
-sudo service docker restart
+# Jenkins builds will install docker here. Gitlab runners have it already in the image
+if ! hash docker 2>/dev/null; then
+    sudo apt-get -qy --force-yes install docker-ce || apt-get -qy --force-yes install docker-ce
+    # make sure Docker uses the block storage to store docker containers.
+    sudo bash -c 'cat << EOF > /etc/docker/daemon.json
+    {
+    "data-root": "/var/lib/docker/",
+    "storage-driver": "overlay2"
+    }
+    EOF'
+    sudo service docker restart
+fi
```

Comment:
This should actually be moved to `jenkins-yoctobuild-init-script.sh` instead.

Reply:
TBH, I don't remember exactly the reason, it was in my early days in Mender, but we did have this in the init script and moved it here for **some** reason, see commit 008eacc

I would like not to touch again the Jenkins init script, as then we would need to rebuild the image in Google Cloud Platform.

Reply:
Lol, ok. I guess it's not that important, and I'm sure there was a reason.